newsroom/search.py

```diff
@@ -513,7 +513,12 @@ def prefill_search_items(self, search):
     def prefill_search_highlights(self, search, req):
         query_string = search.args.get("q")
         query_string_settings = app.config["ELASTICSEARCH_SETTINGS"]["settings"]["query_string"]
-        if app.data.elastic.should_highlight(req) and query_string:
+        advanced_search_query = search.args.get("advanced_search")
+        if (
+            (app.data.elastic.should_highlight(req) and query_string)
+            or advanced_search_query
+            and len(json.loads(advanced_search_query).get("fields", [])) != 0
+        ):
```

Comment:
this should be rather:
```py
if app.data.elastic.should_highlight(req) and (
    query_string or
    (advanced_search and advanced_search.get("all")) or
    (advanved_search and advanced_search.get("any"))
):
```
the fields config is not so important there, it's rather the queries which are important

Reply:
this condition fails when we applied only advanced search,  we did not get `search.args.get("q")`

Reply:
which part fails? there is an `or` between `query_string` and advanced search checks so should be enough to have any of those

Reply:
we got app.data.elastic.should_highlight(req)  = False